ts/Series/SeriesOnPointComposition.ts

```diff
@@ -0,0 +1,576 @@
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type SVGAttributes from '../Core/Renderer/SVG/SVGAttributes';
+
+import Point from '../Core/Series/Point.js';
+import Series from '../Core/Series/Series.js';
+import SeriesRegistry from '../Core/Series/SeriesRegistry.js';
+import SVGRenderer from '../Core/Renderer/SVG/SVGRenderer.js';
+import SVGElement from '../Core/Renderer/SVG/SVGElement.js';
+
+const { bubble, pie, sunburst } = SeriesRegistry.seriesTypes;
+import CU from './CenteredUtilities.js';
+const { getCenter } = CU;
+
+const drawPointsFunctions: { [key: string]: Function; } = {
+    pieDrawPoints: pie.prototype.drawPoints,
+    sunburstDrawPoints: sunburst && sunburst.prototype.drawPoints
+};
+
+const translateFunctions: { [key: string]: Function; } = {
+    pieTranslate: pie.prototype.translate,
+    sunburstTranslate: sunburst && sunburst.prototype.translate
+};
+
+import U from '../Core/Utilities.js';
+import Chart from '../Core/Chart/Chart';
+const {
+    addEvent,
+    defined,
+    find,
+    isNumber
+} = U;
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+
+type BubblePxExtremes = { minPxSize: number; maxPxSize: number };
+type BubbleZExtremes = { zMin: number; zMax: number };
+
+declare module '../Core/Series/SeriesLike' {
+    interface SeriesLike {
+        onPoint?: SeriesOnPointComposition.Additions;
+    }
+}
+
+declare module '../Core/Series/SeriesOptions' {
+    interface SeriesOptions {
+        onPoint?: OnPoint;
+    }
+}
+
+interface OnPoint {
+    connectorOptions?: SVGAttributes;
+    id: string;
+    position?: Position;
+    z?: number;
+}
+
+interface Position {
+    x?: number
+    y?: number;
+    offsetX?: number;
+    offsetY?: number;
+}
+
+/* *
+ *
+ *  Composition
+ *
+ * */
+
+namespace SeriesOnPointComposition {
+
+    /* *
+     *
+     *  Declarations
+     *
+     * */
+
+    export declare class SeriesComposition extends Series {
+        onPoint: Additions;
+    }
+
+    /* *
+     *
+     *  Constants
+     *
+     * */
+
+    const composedClasses: Array<Function> = [];
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    /* eslint-disable valid-jsdoc */
+
+    /**
+     * Extends the series with a small addition.
+     *
+     * @private
+     *
+     * @param SeriesClass
+     * Series class to use.
+     *
+     * @param ChartClass
+     * Chart class to use.
+     */
+    export function compose<T extends typeof Series>(
+        SeriesClass: T,
+        ChartClass: typeof Chart
+    ): (typeof SeriesComposition&T) {
+        if (composedClasses.indexOf(SeriesClass) === -1) {
+            composedClasses.push(SeriesClass);
+
+            const pieProto = pie.prototype;
+
+            pieProto.getCenter = seriesGetCenter;
+            pieProto.drawPoints = seriesDrawPoints;
+            pieProto.translate = seriesTranslate;
```

Comment:
Historically we have done a lot of overwriting prototype functions, but I am wondering if we can avoid this. For `getCenter` and `translate` we can probably do it in a better way by using events. Try looking for existing event handlers that can get the job done in time, otherwise we could also add events for those functions. For the `drawPoints` override we can probably use the `afterRender` event hooks instead.